### PR TITLE
feat(card-selection): add code to propogate the CardSelectionStoreData to the card view model

### PR DESCRIPTION
**Pull request by RobGallo**
#### Description of changes

This change adds the data from the CardSelectionStore into the cards view model. The store data is first transformed into data relevant to the view model by a call to getCardSelectionViewData. The CardSelectionViewData is then passed to getUnifiedRuleResults where it is assigned to newly added fields in the view model data. The data can now be consumed by the details view.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/tests/unit/common/rule-based-view-model-provider.test.ts [hunk 1]
```diff
@@ -92,11 +110,12 @@ describe('RuleBasedViewModelProvider', () => {
...
                     ],
+                    isExpanded: false,
                 } as CardRuleResult,
             ],
         };
 
-        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(rules, results);
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults(rules, results, cardSelectionViewData);
 
         expect(actualResults).toEqual(expectedResults);
```
Comment (Shobhit1) [inline src/tests/unit/common/rule-based-view-model-provider.test.ts]:
```suggestion
        expect(actualResults).toMatchSnapshot();
```

File: src/common/rule-based-view-model-provider.ts [hunk 2]
```diff
@@ -1,15 +1,28 @@
...
+    AllRuleResultStatuses,
+    CardResult,
+    CardRuleResult,
+    CardRuleResultsByStatus,
+    CardRuleResultStatus,
+} from './types/store-data/card-view-model';
 import { UnifiedResult, UnifiedRule } from './types/store-data/unified-data-interface';
 
-export type GetUnifiedRuleResultsDelegate = (rules: UnifiedRule[], results: UnifiedResult[]) => CardRuleResultsByStatus;
+export type GetUnifiedRuleResultsDelegate = (
```
Comment (dbjorge) [inline src/common/rule-based-view-model-provider.ts]: This sounds like it's transforming some other input type into unified rule results, but it seems to actually be transforming unified rules/results into `CardRuleResults`. Maybe this (and the similar nearby functions) would be better named `GetCardRuleResultsDelegate`?

File: src/tests/unit/common/rule-based-view-model-provider.test.ts [hunk 3]
```diff
@@ -125,11 +144,12 @@ describe('RuleBasedViewModelProvider', () => {
         };
     }
 
-    function createUnifiedResultStub(status: InstanceResultStatus, id: string): UnifiedResult {
+    function createUnifiedResultStub(status: InstanceResultStatus, id: string, isSelected: boolean = false): CardResult {
```
Comment (dbjorge) [inline src/tests/unit/common/rule-based-view-model-provider.test.ts]: createCardResultStub?

File: src/tests/unit/common/rule-based-view-model-provider.test.ts [hunk 4]
```diff
@@ -1,104 +1,57 @@
...
+        expect(actualResults).toEqual(null);
+    });
+
+    test('getUnifiedRuleResults with null card selection view data', () => {
+        const actualResults: CardRuleResultsByStatus = getUnifiedRuleResults([], [], null);
 
         expect(actualResults).toEqual(null);
     });
 
     test('getUnifiedRuleResults', () => {
```
Comment (dbjorge) [inline src/tests/unit/common/rule-based-view-model-provider.test.ts]: I think the `isSelected`/`isExpanded` detection in here is the most interesting bit of logic being added by this PR; should there be tests that cover those different combinations?

Comment (Shobhit1) [reply src/tests/unit/common/rule-based-view-model-provider.test.ts]: Updated tests.

**Pull request merged**